common/src/main/java/com/wynntils/screens/base/widgets/BasicTexturedButton.java

```diff
@@ -20,23 +20,59 @@ public class BasicTexturedButton extends WynntilsButton {
     private final Consumer<Integer> onClick;
     private List<Component> tooltip;
 
+    private int renderTooltip;
+    private boolean scaleTexture;
+
+    public static final int RENDER_TOOLTIP_ABOVE = 0;
+    public static final int RENDER_TOOLTIP_BELOW = 1;
+
     public BasicTexturedButton(
             int x, int y, int width, int height, Texture texture, Consumer<Integer> onClick, List<Component> tooltip) {
+        this(x, y, width, height, texture, onClick, tooltip, RENDER_TOOLTIP_ABOVE, false);
+    }
+
+    public BasicTexturedButton(
+            int x,
+            int y,
+            int width,
+            int height,
+            Texture texture,
+            Consumer<Integer> onClick,
+            List<Component> tooltip,
+            int renderTooltip,
+            boolean scaleTexture) {
         super(x, y, width, height, Component.literal("Basic Button"));
         this.texture = texture;
         this.onClick = onClick;
+        this.renderTooltip = renderTooltip;
+        this.scaleTexture = scaleTexture;
         this.setTooltip(tooltip);
     }
 
     @Override
     public void renderWidget(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
-        RenderUtils.drawTexturedRect(poseStack, texture, this.getX(), this.getY());
+        if (scaleTexture)
+            RenderUtils.drawScalingTexturedRect(
+                    poseStack,
+                    texture.resource(),
+                    this.getX(),
+                    this.getY(),
+                    0,
+                    getWidth(),
+                    getHeight(),
+                    texture.width(),
+                    texture.height());
+        else RenderUtils.drawTexturedRect(poseStack, texture, this.getX(), this.getY());
 
```

Comment:
I am not going to comment on every case I find, but please self-review yourself and use braces for all if(-else) statements unless they fit in 1 line and are control flow operations (like return, continue). Thanks. 

(As this is one of our not so well documented code style requirements, I will make sure to document this).